src/Workspaces/Core/Portable/Utilities/IDictionaryExtensions.cs

```diff
@@ -21,14 +23,15 @@ internal static class IDictionaryExtensions
             return value;
         }
 
+        [return: MaybeNull]
         public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
         {
             if (dictionary.TryGetValue(key, out var value))
             {
                 return value;
             }
 
-            return default;
+            return default!;
```

Comment:
Is this necessary because the compiler isn't correctly applying the [MaybeNull]? If so, can we comment to that effect?

Reply:
➡️ `[MaybeNull]` only applies to callers. The attribute _is_ the justification for the suppression operator in this case.

Reply:
@jcouv's comment https://github.com/dotnet/roslyn/pull/37512#commitcomment-34504917 implies this is a point-in-time behavior and is supposed to (eventually) apply to bodies. There's a tracking bug so I'm just asking if we can add a comment to the tracking bug.

Reply:
@jasonmalinowski It's already covered by #25372.